frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -1538,6 +1540,21 @@ void Visitor::visit(const Return* node) {
   }
 }
 
+void Visitor::visit(const Select* node) {
+  checkOtherwiseAfterWhens(node);
+}
+
+void Visitor::checkOtherwiseAfterWhens(const Select* sel) {
+  bool seenOtherwise = false;
+  for(int i = 0; i < sel->numWhenStmts(); i++) {
+    auto when = sel->whenStmt(i);
+    if (seenOtherwise && !when->isOtherwise()) {
+      CHPL_REPORT(context_, WhenAfterOtherwise, sel, sel);
```

Comment:
Consider breaking / returning here to avoid emitting 100 errors if someone adds an `otherwise` to the top of a 100-`when` `select`.